### PR TITLE
Moved things from the Util file

### PR DESCRIPTION
**Pull request by oscargus**
I've moved some methods from the Util file to more suitable places.

The destination was determined by where the method was (most) used or where it seems to fit.

As a result of this I also moved the corresponding tests to a more suitable location.

### CODE REVIEW

File: src/main/java/net/sf/jabref/model/entry/BibtexEntry.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import java.text.ParsePosition;
 import java.text.SimpleDateFormat;
 import java.util.*;
+import net.sf.jabref.logic.util.date.YearUtil;
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/model/entry/BibtexEntry.java]: I will move this to model...

**Comment by oscargus** [comment]
Rebased, but will have to double-check more carefully to see that everything worked out as expected.

**Comment by simonharrer** [comment]
Can you please rebase so that this can be merged in?

**Comment by oscargus** [comment]
I'll do that after #335 is merged as that will probably create new conflicts...

**Comment by oscargus** [comment]
As far as I can tell all tests are still here.

**Pull request merged**